lib/ansible/module_utils/common/file.py

```diff
@@ -159,7 +159,7 @@ def set_lock(self, path, tmpdir, lock_timeout=None):
         self.lockfd = open(lock_path, 'w')
 
         if lock_timeout <= 0:
-            fcntl.flock(self.lockfd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            fcntl.flock(self.lockfd, fcntl.LOCK_EX)
```

Comment:
The docs above state:

```
        :kw lock_timeout:
            Wait n seconds for lock acquisition, fail if timeout is reached.
            0 = Do not wait, fail if lock cannot be acquired immediately,
            Default is None, wait indefinitely until lock is released.
```

As such, it's expected that a lock will return immediately here if it cannot be acquired immediately

If you want it to wait indefinitely, you need to pass `lock_timeout=None` instead of `lock_timeout=0`

Reply:
However, the code may need adjusted to actually support `None`, it looks like it will fail with a value of `None`